File: benchmark_scripts/pi_approx-seq/jl/pi_approx.jl
function approx_pi(num_intervals)
    pi = 0.0
    w = 1.0 / num_intervals
    for i in 0:num_intervals-1
        llocal = (i + 0.5) * w  # local is a keyword in Julia
        pi += 4.0 / (1.0 + llocal * llocal)
    end
    return pi * w
end

# Main
function main()
    if length(ARGS) != 1
        println("Usage: julia pi_approx.jl NUM_INTERVALS")
        exit(1)
    end

    start_data = time_ns()
    num_intervals = parse(Int, ARGS[1])

    start_compute = time_ns()
    sort!(arr, threshold, workers)
    stop_time =time_ns()

    duration_ete = (stop_time - start_data) * 1e-9
    duration_compute = (stop_time - start_compute) * 1e-9

    println("$(duration_ete),$(duration_compute),$(c)")
end

main()

File: benchmark_scripts/pi_approx-seq/py/pi_approx.py
import sys
import time

def approx_pi(num_intervals):
    pi = 0.0
    w = 1.0 / num_intervals                 # step width
    for i in range(num_intervals):          # for each subinterval
        local = (i + 0.5) * w               # local midpoint
        pi += 4.0 / (1.0 + local * local)   # f(local)
    
    return pi * w

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python pi_approx.py NUM_INTERVALS")
        sys.exit(1)

    start_data = time.time()
    num_intervals = int(sys.argv[1])

    start_compute= time.time()
    pi = approx_pi(num_intervals)
    end = time.time()

    duration_ete = end - start_data
    duration_sort = end - start_compute
    print(f'{duration_ete},{duration_sort},{pi:.16f}')
